src/wallet/generateFaucetWallet.ts

```diff
@@ -23,40 +23,45 @@ const MAX_ATTEMPTS = 20; // Maximum attempts to retrieve a balance
 /**
  * Generates a random wallet with some amount of XRP (usually 1000 XRP).
  *
- * @param this
- * @param address - An existing XRPL address to fund, if undefined, a new wallet will be created.
+ * @param client - Client.
+ * @param wallet - An existing XRPL Wallet to fund, if undefined, a new Wallet will be created.
  * @returns A Wallet on the Testnet or Devnet that contains some amount of XRP.
  */
 async function generateFaucetWallet(
-  this: Client,
-  address?: string
-): Promise<FaucetWallet | void> {
-  if (!this.isConnected()) {
+  client: Client,
+  wallet?: Wallet
+): Promise<Wallet | void> {
+  if (!client.isConnected()) {
     throw new RippledError("Client not connected, cannot call faucet");
   }
 
   // Initialize some variables
   let body: Uint8Array | undefined;
   let startingBalance = 0;
-  const faucetUrl = getFaucetUrl(this);
-
-  // If the user provides an existing wallet to fund
-  if (address && isValidAddress(address)) {
-    // Create the POST request body
-    body = new TextEncoder().encode(
-      JSON.stringify({
-        destination: address,
-      })
-    );
-    // Retrieve the existing account balance
-    const addressToFundBalance = await getAddressXrpBalance(this, address);
-
-    // Check the address balance is not undefined and is a number
-    if (addressToFundBalance && !isNaN(Number(addressToFundBalance))) {
-      startingBalance = Number(addressToFundBalance);
-    } else {
-      startingBalance = 0;
-    }
+  const faucetUrl = getFaucetUrl(client);
+
+  // If no existing Wallet is provided or its address is invalid to fund
+  if (!wallet || !isValidAddress(wallet.classicAddress)) {
+    wallet = Wallet.generate();
+  }
+
+  // Create the POST request body
+  body = new TextEncoder().encode(
```

Comment:
AFAICT the `body` could be a `const` defined here. no need to `let`. IMO `let` should be used only when really really necessary

Reply:
Ah you're right, I just noticed this `body` can be refactored to be a `const` now.